platform/machine/azure/flight.go

```diff
@@ -93,8 +94,9 @@ func NewFlight(opts *azure.Options) (platform.Flight, error) {
 			return nil, err
 		}
 
-		_, err = af.api.PrepareNetworkResources(af.ImageResourceGroup)
+		af.Network, err = af.api.PrepareNetworkResources(af.ImageResourceGroup)
 		if err != nil {
+			af.Destroy()
```

Comment:
Do you still need to destroy the network if getting it resulted in failure? I'd say that PrepareNetworkResources should take care of it.

Reply:
I should expand the commit message: if `PrepareNetworkResources` fails because you passed a wrong vnet/subnet name, then you want the resource group + storage account to be deleted, that's why I added the `Destroy()` here.

Reply:
Ah, for some reason I also thought that it was calling Destroy on the just returned network. Nevermind.